File: src/main/java/kr/co/lunatalk/domain/inquiry/domain/InquiryReply.java
package kr.co.lunatalk.domain.inquiry.domain;

import jakarta.persistence.*;
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity;
import kr.co.lunatalk.domain.member.domain.Member;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class InquiryReply extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@OneToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "inquiry_id", nullable = false, unique = true)
	private Inquiry inquiry;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "admin_id", nullable = false)
	private Member admin;

	@Column(nullable = false, columnDefinition = "TEXT")
	private String content;

	@Builder(access = AccessLevel.PRIVATE)
	public InquiryReply(Inquiry inquiry, Member admin, String content) {
		this.inquiry = inquiry;
		this.admin = admin;
		this.content = content;
	}

	public static InquiryReply createReply(Inquiry inquiry, Member admin, String content) {
		return InquiryReply.builder()
			.inquiry(inquiry)
			.admin(admin)
			.content(content)
			.build();
	}

	public void update(String content) {
		this.content = content;
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/inquiry/domain/InquiryReply.kt
package kr.co.lunatalk.domain.inquiry.domain

import jakarta.persistence.*
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity
import kr.co.lunatalk.domain.member.domain.Member

@Entity
open class InquiryReply protected constructor() : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null
        protected set

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inquiry_id", nullable = false, unique = true)
    open var inquiry: Inquiry? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_id", nullable = false)
    open var admin: Member? = null
        protected set

    @Column(nullable = false, columnDefinition = "TEXT")
    open var content: String = ""
        protected set

    private constructor(inquiry: Inquiry, admin: Member, content: String) : this() {
        this.inquiry = inquiry
        this.admin = admin
        this.content = content
    }

    fun update(content: String) {
        this.content = content
    }

    companion object {
        fun createReply(inquiry: Inquiry, admin: Member, content: String): InquiryReply {
            return InquiryReply(inquiry, admin, content)
        }
    }
}
